src/components/checkout/PaymentSelector.tsx

```diff
@@ -9,22 +9,78 @@ import { CommonTitleMixin } from '../common'
 
 const StyledTitle = styled.h1`
   margin-bottom: 0.75rem;
+  line-height: 1.5;
   ${CommonTitleMixin}
 `
 const StyledDescription = styled.div`
   color: var(--gray-dark);
   font-size: 14px;
   letter-spacing: 0.4px;
+  line-height: 1.5;
 `
 
 const PaymentSelector: React.FC<{
   value: PaymentProps | null
   onChange: (value: PaymentProps | null) => void
-}> = ({ value, onChange }) => {
+  isValidating?: boolean
+}> = ({ value, onChange, isValidating }) => {
   const { formatMessage } = useIntl()
   const { settings } = useApp()
   const [selectedPaymentMethod, setSelectedPaymentMethod] = useState<PaymentProps | null>(value)
 
+  const paymentOptions: { payment: PaymentProps; name: string; permissions: boolean }[] = [
+    {
+      payment: { gateway: 'spgateway', method: 'credit' },
+      name: formatMessage(checkoutMessages.label.credit),
+      permissions: settings['payment.spgateway.credit.enable'] === '1',
+    },
+    {
+      payment: { gateway: 'spgateway', method: 'vacc' },
+      name: formatMessage(checkoutMessages.label.vacc),
+      permissions: settings['payment.spgateway.vacc.enable'] === '1',
+    },
+    {
+      payment: { gateway: 'spgateway', method: 'cvs' },
+      name: formatMessage(checkoutMessages.label.cvs),
+      permissions: settings['payment.spgateway.cvs.enable'] === '1',
+    },
+    {
+      payment: { gateway: 'spgateway', method: 'instflag' },
+      name: formatMessage(checkoutMessages.label.instFlag),
+      permissions: settings['payment.spgateway.instflag.enable'] === '1',
+    },
+    {
+      payment: { gateway: 'spgateway', method: 'unionpay' },
+      name: formatMessage(checkoutMessages.label.unionPay),
+      permissions: settings['payment.spgateway.unionpay.enable'] === '1',
+    },
+    {
+      payment: { gateway: 'spgateway', method: 'webatm' },
+      name: formatMessage(checkoutMessages.label.webAtm),
+      permissions: settings['payment.spgateway.webatm.enable'] === '1',
+    },
+    {
+      payment: { gateway: 'spgateway', method: 'barcode' },
+      name: formatMessage(checkoutMessages.label.barcode),
+      permissions: settings['payment.spgateway.barcode.enable'] === '1',
+    },
+    {
+      payment: { gateway: 'paypal', method: 'credit' },
+      name: formatMessage(checkoutMessages.label.paypal),
+      permissions: settings['payment.paypal.credit.enable'] === '1',
+    },
+    {
+      payment: { gateway: 'parenting', method: 'credit' },
+      name: formatMessage(checkoutMessages.label.credit) + `(${formatMessage(checkoutMessages.label.parenting)})`,
+      permissions: settings['payment.parenting.credit.enable'] === '1',
+    },
+    {
+      payment: { gateway: 'commonhealth', method: 'credit' },
+      name: formatMessage(checkoutMessages.label.credit),
+      permissions: settings['payment.commonhealth.credit.enable'] === '1',
+    },
+  ]
+
```

Comment:
payment methods should be got from the database.
this is the public repo, we should not expose the customer info here

```js
keys(settings).filter(key => key.startsWith('payment.')).map(key => {
    const [, gateway, method, ...rest] = key.split('.')
    return {
          payment: { gateway, method },
          name: formatMessage(checkoutMessages.label.credit),  // think about here
    }
})
```

we don't need the permission property. if it is not available, we won't show it

Reply:
still need to filter setting value equal 1 , some payment enable value will be 0